sklearn/tree/tree.py

```diff
@@ -269,6 +277,10 @@ def fit(self, X, y, sample_weight=None, check_input=True):
         min_samples_split = max(self.min_samples_split,
                                 2 * self.min_samples_leaf)
 
+        if X_idx_sorted is None and self.presort == True and not issparse(X):
+            X_idx_sorted = np.asfortranarray(np.argsort(X, axis=0),
+                                             dtype=np.int32)
```

Comment:
It is to make it stateless, and also make it cleaner what is happening. Having a splitter object which remembers a dataset is more difficult to understand than passing in a presorted matrix. It also allowed for parallel trees to be built without a duplication of effort when I thought forests might want to have presorting. I have made a note.